### PR TITLE
Vb-3929 - handle alerts updated events

### PR DESCRIPTION
**Pull request by dhirajshettymoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/integration/domainevents/PrisonVisitsEventsIntegrationTestBase.kt [hunk 1]
```diff
@@ -114,13 +109,7 @@ abstract class PrisonVisitsEventsIntegrationTestBase {
   lateinit var prisonerNonAssociationCreatedNotifier: PrisonerNonAssociationNotifierCreatedNotifier
 
   @SpyBean
-  lateinit var prisonerNonAssociationNotifierClosedNotifier: PrisonerNonAssociationNotifierClosedNotifier
```
[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/integration/domainevents/PrisonVisitsEventsIntegrationTestBase.kt:117]: yes it wasn't being used - we are only handling "non-associations.created" in code and tests.

**Review by adaviesMOJ — APPROVED** [review]
LGTM - Probably worth  updating the title & description of PR

**Review by AledWEvans — DISMISSED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/listeners/events/additionalinfo/PrisonerReceivedInfo.kt [hunk 2]
```diff
@@ -2,12 +2,14 @@ package uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.servic
...
 import com.fasterxml.jackson.annotation.JsonProperty
 import jakarta.validation.constraints.NotBlank
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.visit.scheduler.enums.PrisonerReceivedReasonType
 
 data class PrisonerReceivedInfo(
   @NotBlank
   @JsonProperty("nomsNumber")
   val prisonerNumber: String,
+
   @NotBlank
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/listeners/events/additionalinfo/PrisonerReceivedInfo.kt]: same again

**Review by AledWEvans — COMMENTED** [review]
Some comments

**Review by AledWEvans — DISMISSED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/visit/scheduler/visitnotification/PrisonerReceivedNotificationDto.kt [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
 
 import jakarta.validation.constraints.NotBlank
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.visit.scheduler.enums.PrisonerReceivedReasonType
 import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.service.listeners.events.additionalinfo.PrisonerReceivedInfo
 
 data class PrisonerReceivedNotificationDto(
   @NotBlank
   val prisonerNumber: String,
+
   @NotBlank
```
[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/visit/scheduler/visitnotification/PrisonerReceivedNotificationDto.kt]: Updated. Part of merged PR.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/listeners/events/additionalinfo/PrisonerReceivedInfo.kt (hunk 2 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/listeners/events/additionalinfo/PrisonerReceivedInfo.kt]: Updated

**Review by adaviesMOJ — DISMISSED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/visit/scheduler/visitnotification/PrisonerAlertsAddedNotificationDto.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  @NotBlank
+  val prisonerNumber: String,
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  val alertsAdded: List<String>,
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  val alertsRemoved: List<String>,
+
+  val description: String,
```
Comment (adaviesMOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/visit/scheduler/visitnotification/PrisonerAlertsAddedNotificationDto.kt:17]: does this need a @NotBlank annotation?

Comment (adaviesMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/visit/scheduler/visitnotification/PrisonerAlertsAddedNotificationDto.kt:18]: Implemented

**Review by adaviesMOJ — APPROVED** [review]

**Pull request merged**